source/loop.cpp

```diff
@@ -255,6 +305,11 @@ void Loop::SaveResults(int i,double time)
     results.density[i] = __state[2];
     results.velocity[i] = velocity;
   }
+  
+  if( parameters.use_lookup_table_losses )
+  {
+      parameters.previous_density = __state[2];
+  }
```

Comment:
I was surprised to see this in `SaveResults`. Is there a particular reason that it's here?

Reply:
Good question.  Let me look around, because I can't remember right now why I put this there.

Reply:
Might be a dumb question on my part, but where is the density calculated?  I'm storing the previous time-step's density here to later use for a weighted average.  I'm not seeing the actual calculation.  I'm guessing I placed it here because this is just a function for storing the calculated values.

Reply:
Not a dumb question. The actual calculation of all the quantities (including density) is done internally by the boost integrator(s) and then stored in `state`. If you just always need the previous timestep's density, you can always grab it from `results.density[i-1]` if you know the current time index.

Reply:
Right, got it.  The issue is then that the function that uses this, `CalculateAbundanceFactor`, doesn't have the time index.  So alternatively I could try to pass it the time index, which might be done by storing it in a similar fashion in parameters.  Or is there a clever way to get that index?

Reply:
In that case, I wouldn't worry too much about it. Passing around the time index seems messier.

Reply:
Sounds good.  In the future, if there's another function that needs the time index, then, perhaps it would be good to revisit this.